flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/enumerator/PulsarSourceEnumerator.java

```diff
@@ -175,11 +175,11 @@ public void close() {
     private void seekStartPosition(Set<TopicPartition> partitions) {
         ConsumerBuilder<byte[]> consumerBuilder =
                 createConsumerBuilder(pulsarClient, Schema.BYTES, configuration);
-        Set<String> seekedPartitions = new HashSet<>();
+        Set<String> seekedTopics = new HashSet<>();
 
         for (TopicPartition partition : partitions) {
             String topicName = partition.getFullTopicName();
-            if (seekedPartitions.add(topicName)) {
+            if (!assignmentState.containsTopic(topicName) && seekedTopics.add(topicName)) {
```

Comment:
We could avoid this check by using only hooking into `assign` which only gets the added partitions. Did you try that out? (It might not work out, but it feels the better place).

Reply:
Clarified offline that this would not be as straight-forward and will be eventually superseded by the protocol sketched above.